File: solutions/009/009.py
import math


def find_triplet_product(s):
    r = int(math.sqrt(s))
    for m in range(r, 2, -1):
        for n in range(m, 1, -1):
            a = m ** 2 - n ** 2
            b = 2 * (m * n)
            c = (m * m) + (n * n)
            if a + b + c == s:
                print(a*b*c)
                return


find_triplet_product(1000)

File: solutions/009/_009.cs
using System;

public class _009 {
    public static void Main(string[] args) {
        int s = 1000;
        int r = (int) Math.Sqrt(s);
        for (int m = r; m > 1; m--)
            for (int n = m; n > 2; n--) {
                int a = m * m - n * n;
                int b = 2 * (m * n);
                int c = m * m + n * n;
                if (a + b + c == s) {
                    Console.WriteLine(a * b * c);
                    return;
                }
            }
    }
}
